src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java

```diff
@@ -151,13 +153,32 @@ public Object instanceMainPostParseArgs() {
 
         LoggingUtils.setLoggingLevel(VERBOSITY);  // propagate the VERBOSITY level to logging frameworks
 
-        for (final File f : TMP_DIR) {
+        for (final String s : TMP_DIR) {
+            final Path p = IOUtils.getPath(s);
             // Intentionally not checking the return values, because it may be that the program does not
             // need a tmp_dir. If this fails, the problem will be discovered downstream.
-            if (!f.exists()) f.mkdirs();
-            f.setReadable(true, false);
-            f.setWritable(true, false);
-            System.setProperty("java.io.tmpdir", f.getAbsolutePath()); // in loop so that last one takes effect
+            if (!Files.exists(p)) {
+                try {
+                    Files.createDirectories(p);
+                } catch (IOException e) {
+                    // intentionally ignoring
+                }
+            }
+            try {
+                // add the READ/WRITE permissions to the actual permissions of the file
+                final Set<PosixFilePermission> permissions = Files.getPosixFilePermissions(p);
+                // only set the permissions if they change
+                if (permissions.addAll(Arrays.asList(
+                        PosixFilePermission.OWNER_READ, PosixFilePermission.GROUP_READ, PosixFilePermission.OTHERS_READ,
+                        PosixFilePermission.OWNER_WRITE, PosixFilePermission.GROUP_WRITE, PosixFilePermission.OTHERS_WRITE))) {
+                    Files.setPosixFilePermissions(p, permissions);
```

Comment:
Can you get rid of this code to change the permissions on the temp dir, and throw a `UserException` instead if the temp dir is not writable/readable?

Reply:
Done, although in that case is a change of behavior - if the temp dir is not used at all, this can fail for every tool...

Reply:
I think we always use the temp dir in GATK -- if for nothing else, then to extract and load the GKL library for compression/decompression.

Reply:
Is the GKL library _**always**_  extracted, even if not used at all? I don't think that it is (or should be) like that...